bin/wasm-node/javascript/src/index.js

```diff
@@ -16,29 +16,33 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import { Worker, workerOnError, workerOnMessage } from './compat-nodejs.js';
+import { v4 as uuidv4 } from 'uuid';
 
 export class SmoldotError extends Error {
   constructor(message) {
     super(message);
   }
 }
 
+function defaultLog(level, target, message) {
+  const fmt = (target, message) => `[${target}] ${message}`;
+  if (level <= 1) {
+    console.error(fmt(target, message));
+  } else if (level == 2) {
+    console.warn(fmt(target, message));
+  } else if (level == 3) {
+    console.info(fmt(target, message));
+  } else if (level == 4) {
+    console.debug(fmt(target, message));
+  } else {
+    console.trace(fmt(target, message));
+  }
+}
+
 export async function start(config) {
   config = config || {};
 
-  const logCallback = config.logCallback || ((level, target, message) => {
```

Comment:
I don't think extracting this improves the code readability.

Reply:
You were creating the callback inside the function which potentially leads to multiple copies of the same function

Reply:
Is there a downside in having multiple copies of the same function?

Reply:
its not just the function it's a closure so its references to everything in its outerscope.  I dont feel strongly about it though but in general avoiding creating closures where they're not needed is a considered a good practice in javascript because that's the most common source of memory leaks.